qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -1090,4 +1100,52 @@ private ResolvedJavaTypeInfo validateIterable(Part part, Section ownerSection,
 		}
 		return resolvedJavaType;
 	}
-}
+
+	private void validateSwitchWhenPart(JavaMethodInfo method, JavaMemberInfo member, Section ownerSection,

```

Comment:
JavaMethodInfo is a JavaMemeberInfo so I think you can remove method parzmeter znd use only member parameter